src/network.toit

```diff
@@ -224,7 +225,14 @@ class EndpointHttp implements Endpoint:
           image = flash-image request.content-length request.body container-name defines --crc32=crc32
           respond-ok writer
         run-message := path == "/install" ? "installed and started" : "started"
-        run-image image run-message container-name defines
+        // We don't run the image from within the request-mutex, as we need to be able to
+        // shut down the server while trying to run the image.
+        // For the same reason we must make sure that the task that runs the image isn't
+        // the server task.
+        if Task.current == server-task:
```

Comment:
I don't quite understand this. If the `server` isn't created with a `--max-tasks` setting higher than 1, then this lambda will always be called on the `server-task`. So in practice, you'll always have to create another task here.

But didn't you change `run-image` so it returns immediately?

Reply:
The `run-image` was potentially blocking to shut down the wifi.
Incidentally, the uart was blocked this way too.

Now the `run-image` creates a task if it might need to block.
done.